implementations/rust/ockam/ockam_transport_tcp/src/portal/outlet_listener.rs

```diff
@@ -81,34 +107,84 @@ impl Worker for TcpOutletListenWorker {
         let their_identifier = SecureChannelLocalInfo::find_info(msg.local_message())
             .map(|l| l.their_identifier())
             .ok();
-        let src_addr = msg.src_addr();
+
+        let src_addr = msg.src_addr().clone();
         let msg = msg.into_local_message();
-        let return_route = msg.return_route;
-        let body = msg.payload;
-        let msg = PortalMessage::decode(&body)?;
 
-        if !matches!(msg, PortalMessage::Ping) {
-            return Err(TransportError::Protocol)?;
+        let remote_address = msg.return_route.recipient()?.clone();
+
+        let map_key = MapKey {
+            identifier: their_identifier.clone(),
+            remote_address,
+        };
+
+        if self.options.skip_handshake {
+            let sender_remote = self
+                .outlet_registry
+                .started_workers
+                .read()
+                .unwrap()
+                .get(&map_key)
+                .cloned();
+            if let Some(sender_remote) = sender_remote {
+                return Self::reroute_msg(ctx, sender_remote.clone(), msg).await;
```

Comment:
`sender_remote.clone()` ← this clone should be avoidable

Reply:
I hope compiler is clever enough to not clone it, but let's not rely on that :)